lib/storage.js

```diff
@@ -53,6 +53,17 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
+            if (protocol === DATA_PROTOCOL) {
+                await datauri.download(assetReference, source.path);
+                if(!fileExistsAndIsNotEmpty(source.path)){
+                    throw new SourceUnsupportedError(`Invalid or missing local file ${source.path}`);
+                }
+                const preSignedUrl = await datauri.getPreSignedUrl(source.path);
+                console.log(`Uploaded data URI content to storage and generated presigned url`);
```

Comment:
`datauri.getPreSignedUrl` could throw error for `temporaryCloudStorage.upload / temporaryCloudStorage.generatePresignURL` - we should handle it and also have a test case. This is a common flow and an error here could cause regression. So its good to confirm.

Reply:
Here the failure would eventually bubble up.
The point is that if we don't get the URL, we should stop execution early and not even attempt to contact services to which we could not even send all their needed params (since one of them, the presigned URL of the source, would be missing since generation failed). A lot of services don't handle data uris directly, so not being able to turn it into something else will be an issue here.

@pheenomenon What do you mean with handling the error, then?

Reply:
Sounds good, if bubbling up won't result any new regressions.

Reply:
@pheenomenon @tmathern Added Error handling for preSignedUrl and threw a meaningful error.